toolkit/scripts/check_required_kernel_configs.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for required kernel configs in a kernel config file
+# Usage: python3 check_required_kernel_configs.py --required_configs <path to json of required configs> --config_file <path to config being checked>
+
+import json
+import argparse
+import sys
+from kernel_sources_analysis import get_data_from_config, extract_kernel_dir_name, extract_config_arch
+
+# Define a class
+    #incorrect configs is map: {config: (newValue, expectedValue, comment, PR)}
+class IncorrectConfig:
+    def __init__(self, name, newValue, expectedValue, comment, PR):
+        self.name = name
+        self.newValue = newValue
+        self.expectedValue = expectedValue
+        self.comment = comment
+        self.PR = PR
+
+def check_strings_in_file(json_file, kernel, arch, input_config_data):
```

Comment:
>check_strings_in_file [](http://example.com/codeflow?start=4&length=21)

I think we could be more specific in naming this function. I think some complex systems could be briefly described as "checking strings in files". What I'm getting at: we're not getting much info about the purpose and potential result of calling this function from the current name.;)